modeshape-jcr/src/main/java/org/modeshape/jcr/cache/document/DocumentTranslator.java

```diff
@@ -868,15 +868,22 @@ protected long insertChildren( EditableDocument document,
 
     public ChildReferences getChildReferences( WorkspaceCache cache,
                                                Document document ) {
+        Name primaryType = getPrimaryType(document);
+        NodeTypes nodeTypes = getNodeTypes(cache);
+
+        boolean isUnorderedCollection = nodeTypes != null && nodeTypes.isUnorderedCollection(primaryType);
         boolean hasChildren = document.containsField(CHILDREN);
         boolean hasFederatedSegments = document.containsField(FEDERATED_SEGMENTS);
-        if (!hasChildren && !hasFederatedSegments) {
+        if (!hasChildren && !hasFederatedSegments && !isUnorderedCollection) {
             return ImmutableChildReferences.EMPTY_CHILD_REFERENCES;
         }
+
+        if (isUnorderedCollection) {
+            // the base 'mode:unordered' node types do not contain a PRB field...
+            return document.containsField(PRB) ?  new BucketedChildReferences(document, this) : ImmutableChildReferences.EMPTY_CHILD_REFERENCES; 
```

Comment:
> I was trying to explain that the nodes types instances (e.g. `jcr:system/jcr:nodeTypes/mode:unorderedTinyCollection`) is at runtime, an unordered collection instance which will/can never have a `$prb` field because it's the actual node type instance (sort of like the meta-data). 

Really? Isn't every child node under `jcr:system/jcr:nodeTypes` an instance of `jcr:nodeType` (or something like that), which does not extend `mode:unorderedCollection`? Or, does this happen when `nodeTypes` is null? If that is the case, then I might change the logic to be something like:

        if (isUnorderedCollection) {
            if ( document.containsField(PRB) ) {
                return new BucketedChildReferences(document, this);
            }
            // Otherwise, we're in startup mode ...
            assert nodeTypes == null;
            return ImmutableChildReferences.EMPTY_CHILD_REFERENCES; 
        }

> but I can add more detail :)

Yes, please add more detail. :-) I had no idea what this code or comment was trying to signify.